pywr/recorders/_recorders.pyx

```diff
@@ -1522,6 +1522,140 @@ cdef class MinimumThresholdVolumeStorageRecorder(BaseConstantStorageRecorder):
 MinimumThresholdVolumeStorageRecorder.register()
 
 
+cdef class DailyCountIndexParameterRecorder(IndexParameterRecorder):
+    def __init__(self, model, IndexParameter parameter, int threshold, *args, **kwargs):
+        super(DailyCountIndexParameterRecorder, self).__init__(model, parameter, *args, **kwargs)
+        self.threshold = threshold
+
+    cpdef setup(self):
+        self._count = np.zeros(len(self.model.scenarios.combinations), np.int32)
+
+    cpdef reset(self):
+        self._count[...] = 0
+
+    cpdef after(self):
+        cdef Timestep ts = self.model.timestepper.current
+
+        for scenario_index in self.model.scenarios.combinations:
+            value = self._param.index(ts, scenario_index)
+            if value >= self.threshold:
+                # threshold achieved, increment count
+                self._count[scenario_index.global_id] += 1
+
+    cpdef double[:] values(self):
+        return np.asarray(self._count).astype(np.float64)
+DailyCountIndexParameterRecorder.register()
+
+
+cdef class AnnualCountIndexThresholdRecorder(Recorder):
+    """
+    For each scenario, count the number of times a parameter exceeds a threshold in each year.
```

Comment:
This says "a parameter", but takes a list of parameters.

Should this recorder supporting the `values()` method and aggregate over the parameters?